dataflow/gemma-flex-template/README.md

```diff
@@ -135,38 +129,41 @@ class FormatOutput(beam.DoFn):
 ## Build the Flex Template
 Run the following code from the directory to build the Dataflow flex template.
 
-- Replace `$TEMPLATE_FILE` with a Google Cloud Storage path written as a JSON file.
+- Replace `$GCS_BUCKET` with a Google Cloud Storage bucket.
 - Set `SDK_CONTAINER_IMAGE` to the name of the Docker image created previously.
 - `$PROJECT` is the Google Cloud project that you created previously. 
 
 ```sh
-gcloud dataflow flex-template build $TEMPLATE_FILE \
+gcloud dataflow flex-template build gs://$GCS_BUCKET/config.json \
   --image $SDK_CONTAINER_IMAGE \
   --sdk-language "PYTHON" \
   --metadata-file metadata.json \
   --project $PROJECT
 ```
 
 ## Start the pipeline
-To start the Dataflow streaming job, run the following code from the directory. Replace `$TEMPLATE_FILE`, `$REGION`, `$GCS_BUCKET`, `$INPUT_SUBSCRIPTION`, `$OUTPUT_TOPIC`, `$SDK_CONTAINER_IMAGE`, and `$PROJECT` with the Google Cloud project resources you created previously. It might take as much as 15 minutes for the worker to start up and to begin accepting messages from the input Pub/Sub topic. 
+To start the Dataflow streaming job, run the following code from the directory. Replace `$TEMPLATE_FILE`, `$REGION`, `$GCS_BUCKET`, `$INPUT_SUBSCRIPTION`, `$OUTPUT_TOPIC`, `$SDK_CONTAINER_IMAGE`, and `$PROJECT` with the Google Cloud project resources you created previously. Ensure that `$INPUT_SUBSCRIPTION` and `$OUTPUT_TOPIC` are the fully qualified subscription and topic names, respectively. It might take as much as 30 minutes for the worker to start up and to begin accepting messages from the input Pub/Sub topic. 
 
 ```sh
 gcloud dataflow flex-template run "gemma-flex-`date +%Y%m%d-%H%M%S`" \
   --template-file-gcs-location $TEMPLATE_FILE \
   --region $REGION \
-  --staging-location $GCS_BUCKET \
+  --temp-location gs://$GCS_BUCKET/tmp \
+  --staging-location gs://$GCS_BUCKET \
   --parameters messages_subscription=$INPUT_SUBSCRIPTION \
   --parameters responses_topic=$OUTPUT_TOPIC \
   --parameters device="GPU" \
-  --parameters sdk_container_image=$SDK_CONTAINER_IMAGE \
-  --parameters dataflow_service_options="worker_accelerator=type:nvidia-l4;count:1;install-nvidia-driver:5xx" \
+  --sdk_container_image=$SDK_CONTAINER_IMAGE \
+  --dataflow_service_options="worker_accelerator=type:nvidia-l4;count:1;install-nvidia-driver:5xx" \
```

Comment:
These parameters aren't recognised by gcloud, and aren't described in https://cloud.google.com/sdk/gcloud/reference/dataflow/flex-template/run

```

ERROR: (gcloud.dataflow.flex-template.run) unrecognized arguments:
  --sdk_container_image=us-central1-docker.pkg.dev/$PROJECT_ID/docker/dataflow/gemma-flex:1
  --dataflow_service_options=worker_accelerator=type:nvidia-l4;count:1;install-nvidia-driver:5xx
  To search the help text of gcloud commands, run:
  gcloud help -- SEARCH_TERMS
```

Reply:
> Interesting, we don't expose two pretty important service options for running GPU pipelines via flex templates then. I should be able to update the parameters to pass them through, but that's surprising

* According to https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/dataflow/flex-templates/pipeline_with_dependencies#run-the-template, --parameters sdk_container_image works
* According to https://cloud.google.com/dataflow/docs/reference/service-options , `--additional-experiments` should be used for service options.

Reply:
the former is true, the latter doesn't seem to work? dataflow gets the correct information in the UI but this error is thrown on startup:

![image](https://github.com/GoogleCloudPlatform/python-docs-samples/assets/34928439/4a090b62-b5bf-4c15-84c9-3a69fc8e544c)

This looks like a known issue based on https://github.com/google/dataflow-ml-starter/blob/main/flex/metadata.json

Reply:
Discussed offline;
Can we document the reason for using 5xx? Is it a requirement for L4?

This should be Dataflow docs, but  I am not seeing this configuration option documented in https://cloud.google.com/dataflow/docs/gpu/use-gpus#drivers  or in https://cloud.google.com/dataflow/docs/gpu/gpu-support#availability

Reply:
IIRC back when I was first implementing Gemma-on-Dataflow the 5xx driver was necessary for L4s. Strange to not see any mention of 4xx vs 5xx on https://cloud.google.com/dataflow/docs/reference/service-options, that feels like a gap

Reply:
I think the 5xx drivers are now the default, so we don't need to specify to use them.

Reply:
I'll do a test run locally to see if that's the case

Reply:
Confirmed that 5xx is no longer necessary